### PR TITLE
[ENH] Add minirocket_multivariate_variable for unequal length time series

### PR DESCRIPTION
**Pull request by michaelfeil**
#### Reference Issues/PRs
Refers to #3769 .

#### What does this implement/fix? Explain your changes.
This PR adds an unequal length, multivariate time series https://github.com/angus924/minirocket/blob/main/code/minirocket_multivariate_variable.py.
The MinirocketMultivariateVariable works similar to the MinirocketMultivariate, however can process multivariate timeseries with variable length, therefore requires an own Sktime interface. 

#### Does your contribution introduce a new dependency? If yes, which one?
No.

#### What should a reviewer concentrate their feedback on?
As this is my first contribution, give a broad feedback:
- if the class is implemented correctly, and has sufficient tests
- if the formatting is correct (check doc strings)
- is the doc update sufficient (did not find more for other models)

#### Any other comments?
- also updated the notebook, where the `RidgeClassifierCV` has a deprecation warning of `normalizer=True`
- 
##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/.all-contributorsrc).
- [x] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/sktime/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [x] I've added unit tests and made sure they pass locally. (Python 3.10 conda env, wsl2 )
- [x]The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.

##### For new estimators
- [x] I've added the estimator to the online documentation.
- [x] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Comment by MatthewMiddlehurst** [comment]
Can look at this tomorrow. There is currently a conflict with main, however.

**Comment by michaelfeil** [comment]
> Can look at this tomorrow. There is currently a conflict with main, however.

should be solved, was just the contributors.

File: sktime/transformations/panel/rocket/_minirocket_multivariate_variable.py
Comment (MatthewMiddlehurst) [inline sktime/transformations/panel/rocket/_minirocket_multivariate_variable.py:78]: These should be consistent with the other transformers.

File: sktime/transformations/panel/rocket/_minirocket_multivariate_variable.py
Comment (MatthewMiddlehurst) [inline sktime/transformations/panel/rocket/_minirocket_multivariate_variable.py:35]: [This](https://sktime--3786.org.readthedocs.build/en/3786/api_reference/auto_generated/sktime.transformations.panel.rocket.MiniRocketMultivariateVariable.html#sktime.transformations.panel.rocket.MiniRocketMultivariateVariable) is the generated docs for the estimator. There are a couple of issues, but the other transformers also have them. Not a requirement to fix, but making you aware.

File: examples/minirocket.ipynb
Comment (MatthewMiddlehurst) [inline examples/minirocket.ipynb:56]: This comment seems incorrect.

File: examples/minirocket.ipynb
Comment (MatthewMiddlehurst) [inline examples/minirocket.ipynb:136]: Is this something recommended for all classifiers? I know the default RidgeClassifierCV normalises, but do they recommend it for others as well?

**Comment by MatthewMiddlehurst** [comment]
FYI @angus924 in case you want to comment on this also.

File: sktime/transformations/panel/rocket/_minirocket_multivariate_variable.py
Comment (michaelfeil) [reply sktime/transformations/panel/rocket/_minirocket_multivariate_variable.py:78]: indeed, the ` "scitype:transform-output": "Panel"`. Can be modified to to `primitive`, like in the other rockets.

`"scitype:transform-input"` can't be modified to numpy3d, as the input data is unequal length time series data (+multivariate). Therefore, this minirocket needs an extra class from sktime. 
MiniRocket / MiniRocketMultivariate strictly maybe do not, however, there is a significant amount of different numba code as well.

File: sktime/transformations/panel/rocket/_minirocket_multivariate_variable.py
Comment (michaelfeil) [reply sktime/transformations/panel/rocket/_minirocket_multivariate_variable.py:35]: Okay, should be possible to fit!

File: examples/minirocket.ipynb
Comment (michaelfeil) [reply examples/minirocket.ipynb:136]: True, was not meant to be an editorial recommendation. However, `RidgeClassifierCV(normalize=True)` has now a deprecated `normalize` arg, which is why we may use e.g. StandardScaler() in this tutorial for forward-compatability.

As in https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.RidgeClassifierCV.html
`Deprecated since version 1.0: normalize was deprecated in version 1.0 and will be removed in 1.2. `

**Comment by michaelfeil** [comment]
> FYI @angus924 in case you want to comment on this also.
It would be great to have his approval + comment on code before merging it to this library.

**Comment by angus924** [comment]
Hi, sorry, a bit late to this.

You definitely don't need my approval!

I only have a couple of trivial comments, sorry if I've missed anything.

> Generally for all the ROCKET stuff, I'm not really a fan of the new transformer for each capability approach.

I agree.  The problem is the underlying implementation...  For example, the multivariate implementation is (or, at least, was) meaningfully slower than the univariate implementation.  I have never been able to work out why (although it's been quite a while since I looked at it properly).  Hence, we ended up with two versions: the "canonical" univariate version, and the multivariate version which is ok but flawed in some inscrutable way.  Likewise with variable-length input, it requires handling and iterating over the input data differently (you might even describe it as a "hack"), so you end up with another version again.

If there's a way to get it working with a single backend without compromising efficiency too much, that would be great.  Maybe the different versions are not as different in terms of runtime as I remember.

In any case, I'm happy to go with whatever you think makes sense.

Re the ridge classifier... this is a strange one.  The sklearn implementation changed, deprecating the `normalize` argument, and they suggest using StandardScaler instead.  The odd thing is, StandardScaler does something different to what the `normalize` argument did: `normalize` scales by the L2 norm, `StandardScaler` scales by the standard deviation (although I think these quantities should be within a constant factor of each other?).  It shouldn't make a whole lot of difference in the broad scheme of things, but technically you need to scale by the L2 norm to reproduce the behaviour of the deprecated `normalize` argument.

**Comment by MatthewMiddlehurst** [comment]
I can take a look at some point. If they make a difference that's fine, efficient code for these capabilities is better. If that's the case, we could possibly make these private and create a public facing class which selects an implementation based on the input data?

You can set options for `StandardScaler`, the classification package classifier uses `StandardScaler(with_mean=False)`.

File: examples/minirocket.ipynb
Comment (MatthewMiddlehurst) [reply examples/minirocket.ipynb]: I think the note is fine now. We should probably use `StandardScaler(with_mean=False)` to keep it consistent with the classifier, however.

File: sktime/transformations/panel/rocket/_minirocket_multivariate_variable.py
Comment (MatthewMiddlehurst) [reply sktime/transformations/panel/rocket/_minirocket_multivariate_variable.py:84]: These tags should not change the output at all, the mtype tag for df-list is what sets the input datatype.

It is slightly odd but current convention is to set the tags
```
        "scitype:transform-input": "Series",
        "scitype:transform-output": "Primitives",
```
for transformers which take time series as an input and output primitives (even if the input is a panel).

If I'm wrong and this does actually impact the transformer, let me know.

**Comment by TonyBagnall** [comment]
I'm fine with this going in once ready, but I agree medium term we need to find a way to unify them, I am just tweaking the docstrings and was scratching my head as to why we had duplicates! so I think we probably need an issue to unify the rocket transformers without compromising efficiency. If no one else solves it we can take a look, probably next year though

File: sktime/transformations/panel/rocket/_minirocket_multivariate_variable.py
Comment (michaelfeil) [reply sktime/transformations/panel/rocket/_minirocket_multivariate_variable.py]: Indeed, now it works!

**Comment by michaelfeil** [comment]
Should be all done now!

**Review by MatthewMiddlehurst — APPROVED** [review]
I am happy with the changes made. Will give it a bit to see if anyone else has input, then merge if not.

**Pull request merged**
